include/swift/AST/DiagnosticsSema.def

```diff
@@ -1024,8 +1024,13 @@ NOTE(note_init_parameter,none,
 
 
 ERROR(missing_nullary_call,none,
-     "function produces expected type %0; did you mean to call it with '()'?",
-     (Type))
+     "expected an expression of type %0, but expression has function type "
+     "%1; did you mean to call the function with '()'?",
```

Comment:
Nit: I actually like `it` here. It seems more vague at first blush, but I parse the current sentence has referring to the same subject (the errant expression) by two separate descriptors "the expression" and "the function".